contracts/base/Loans.sol

```diff
@@ -232,6 +230,9 @@ contract Loans is Base, LoansInterface {
             request.duration,
             termsExpiry
         );
+        if (msg.value > 0) {
+            emit CollateralDeposited(loanID, request.borrower, msg.value);
```

Comment:
Why has this moved from above?

Reply:
It is due to how the backend (BE) processes the events. When the LoanTermsSet event is received, the BE creates the Loan/LoanTerms entities. So, then when the CollateralDeposited event is received, the BE can get the Loan entity (by LoanID).
If the CollateralDeposited is emitted first (before LoanTermsSet event), the BE will receive data from a loan that wasn't created yet.